File: 2015/1. Not Quite Lisp/part2.py


with open('input.txt','r') as f:
    s = f.readline().strip()


result = 0
pos = 0
i = 0


while pos != -1:
    paren = s[i]
    pos += 1 if paren == '(' else -1 if paren == ')' else 0
    result += 1
    i += 1
        

print(result)

File: 2015/1. Not Quite Lisp/part2.cs

string s = File.ReadLines("input.txt").First();

int result = 0, pos = 0, i = 0;

while(pos != -1)
{
    char paren = s[i];

    if(paren == '(') pos++;
    else if(paren == ')') pos--;

    result++;
    i++;
}


Console.WriteLine(result);
